### PR TITLE
Code for Query 03

### PR DESCRIPTION
**Pull request by chitralverma**
- Added a readme.md for other new users like me
 - Added a module for scan data sets part
 - Added q03 code
 - Added some utlities to get datasets and validate answers

### CODE REVIEW

File: polars_queries/q03.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from linetimer import linetimer
+
+from polars_queries import datasets
+
+
+@linetimer(name="Query 03", unit='s')
+def q03():
+    customer_ds = (datasets.get_customer_ds()
+                   .filter(pl.col("c_mktsegment") == "HOUSEHOLD")
+                   .select(["c_custkey", "c_mktsegment"])
```
Comment (ritchie46) [inline polars_queries/q03.py:11]: This is immediately a proper test for the optimizer.

You should not have to define this `selection`. Polars should be able to define which columns to select by itself.

So removing these lines should be as fast as with these lines.

Comment (ritchie46) [reply polars_queries/q03.py]: > for this, it seems like my brain optimizer ran before the polars optimizer

LOL

> same except for 1 additional FAST PROJECTION in my version

That should be fine.

File: polars_queries/q03.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import polars as pl
+from linetimer import linetimer
+
+from polars_queries import datasets
+
+
+@linetimer(name="Query 03", unit='s')
+def q03():
+    var1 = var2 = pl.datatypes.datetime(1995, 3, 15)
```
Comment (ritchie46) [inline polars_queries/q03.py:9]: It is preferred to use pythons native `datetime`.

File: polars_queries/q03.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          .filter(pl.col("l_shipdate") > var1)
+          .with_column((pl.col("l_extendedprice") * (1 - pl.col("l_discount"))).alias("revenue"))
+          .groupby(["o_orderkey", "o_orderdate", "o_shippriority"])
+          .agg([pl.sum("revenue").round(2)])
+          .select([pl.col("o_orderkey").alias("l_orderkey"), "revenue", "o_orderdate", "o_shippriority"])
+          .sort(by=["revenue", "o_orderdate"], reverse=[True, False])
+          .limit(10)
+          )
+
+    print(df.collect())
```
Comment (ritchie46) [inline polars_queries/q03.py:30]: Can we add an assert to the expected result?

Comment (chitralverma) [reply polars_queries/q03.py]: done!

File: polars_queries/q3.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+from linetimer import linetimer
+
+from polars_queries import utils
+
+Q_NUM = 3
+
+
+@linetimer(name=f"Overall execution of Query {Q_NUM}", unit='s')
+def q3():
+    var1 = var2 = "1995-03-15"
```
Comment (ritchie46) [inline polars_queries/q3.py:12]: This should be python `datetime`.

File: polars_queries/utils.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+__default_dataset_base_dir = "tables_scale_1"
+__default_answers_base_dir = "tpch-dbgen/answers"
+
+
+def __scan_parquet_ds(path: str):
+    return pl.scan_parquet(path)
+
+
+def get_query_answer(query: int, base_dir: str = __default_answers_base_dir) -> pl.LazyFrame:
+    answer_ldf = pl.scan_csv(join(base_dir, f"q{query}.out"), sep="|", has_header=True)
```
Comment (ritchie46) [inline polars_queries/utils.py:15]: Best to read with `parse_dates=True`

**Comment by ritchie46** [comment]
Thanks @chitralverma

**Pull request merged**